docs/oracle-demo.md

```diff
@@ -80,16 +80,130 @@ Finally, certify the key for use by the connections that request that:
 aea issue-certificates
 ```
 
+### Create the oracle client AEA
+
+From a new terminal (in the same top-level directory), fetch the AEA that will deploy the oracle client contract and call the function that requests the coin price from the oracle contract.
+
+``` bash
+aea fetch fetchai/coin_price_oracle_client:0.1.0
+cd coin_price_oracle_client
+aea install
+```
+
+<details><summary>Alternatively, create from scratch.</summary>
+<p>
+
+Create the AEA that will deploy the contract.
+
+``` bash
+aea create coin_price_oracle
+cd coin_price_oracle
+aea add connection fetchai/http_client:0.16.0
+aea add connection fetchai/ledger:0.12.0
+aea add skill fetchai/simple_oracle_client:0.3.0
+aea install
+aea config set agent.default_connection fetchai/ledger:0.12.0
+```
+
+Then update the agent configuration with the default routing:
+``` bash
+aea config set --type dict agent.default_routing \
+'{
+"fetchai/contract_api:0.10.0": "fetchai/ledger:0.12.0",
+"fetchai/http:0.11.0": "fetchai/http_client:0.16.0",
+"fetchai/ledger_api:0.9.0": "fetchai/ledger:0.12.0"
+}'
+```
+
+Change the default ledger:
+``` bash
+aea config set agent.default_ledger ethereum
+```
+
+</p>
+</details>
+
+Create the private key for the oracle client AEA. Generate and add a key for Ethereum use:
+
+``` bash
+aea generate-key ethereum
+aea add-key ethereum ethereum_private_key.txt
+```
+
 The oracle AEAs require either a locally running test node or a connection to a remote testnet.
 
 ### Setting up with a local Ganache node
 
-The easiest way to test the oracle agents is to set up a local Ethereum test node using Ganache. This can be done by running the following docker command from the directory you started from (in a separate terminal). This command will also fund the account of the AEA:
+The easiest way to test the oracle agents is to set up a local Ethereum test node using Ganache. This can be done by running the following docker command from the directory you started from (in a new terminal). This command will also fund the accounts of the AEAs:
+``` bash
+docker run -p 8545:8545 trufflesuite/ganache-cli:latest --verbose --gasPrice=0 --gasLimit=0x1fffffffffffff --account="$(cat coin_price_oracle/ethereum_private_key.txt),1000000000000000000000" --account="$(cat coin_price_oracle_client/ethereum_private_key.txt),1000000000000000000000"
+```
+
+<details><summary>Run the enclosed Python script (with web3 installed) from the top-level directory to deploy a mock Fetch ERC20 contract and give some test FET to the client agent.</summary>
+<p>
+
+```python
```

Comment:
For now this is ok. But ideally we have this in an agent.

Reply:
Sure, I thought about including the ability to deploy the mock ERC20 contract in the `simple_oracle_client` skill but then it seemed a strange functionality to have, especially when thinking about adding all the arguments to the mock ERC20 constructor as config. But I suppose these could be hard-coded since it is only for testing purposes?

Reply:
The other complicated part is that the address of the ERC20 contract is an argument to the constructor of the oracle contract, so if implemented as above, the oracle agent would need to get the ERC20 address from the client agent and the client agent would then need to get the oracle contract address from the oracle agent.